torch/csrc/autograd/engine.cpp

```diff
@@ -772,6 +773,7 @@ static variable_list call_function(
 
   if(has_post_hooks){
     // NOLINTNEXTLINE(bugprone-use-after-move)
+    at::ThreadLocalStateGuard guard(graph_task->thread_locals_);
```

Comment:
This is not needed I think. This guard is already set at a higher level in the stack here: https://github.com/pytorch/pytorch/blob/842a831f53bc9305c0553f3b2096afc6c1c6efe4/torch/csrc/autograd/engine.cpp#L809-L815

Reply:
Yes. This is duplicated.